File: Lab2/PyLab2/my_file.py
import pickle


def fill_file(path, is_clearing, data):
    info = []

    if not is_clearing:
        with open(path, "rb") as file:
            info = pickle.load(file)

    for call in data:
        info.append(call)

    with open(path, "wb") as file:
        pickle.dump(info, file)


def read_file(path):
    info = []
    with open(path, "rb") as file:
        info = pickle.load(file)

    return info


def display_file(path, caption="File: "):
    print(caption)

    with open(path, "rb") as file:
        info = pickle.load(file)

        for call in info:
            for value in call.values():
                print(value, end=' ')
            print()

File: Lab2/CSharpProgLab2/ProgLab2/MyFile.cs
using System;
using System.Collections.Generic;
using System.IO;
using InputOperations;

namespace FileOperations
{
    public class MyFile
    {
        public static void FillFile(string path, bool isClearing, Call[] data)
        {
            if (isClearing)
                File.Delete(path);

            BinaryWriter binaryWriter = new BinaryWriter(File.Open(path, FileMode.Append));
            for (int i = 0; i < data.Length; i++)
            {
                binaryWriter.Write(data[i].GetPhoneNumber());
                binaryWriter.Write(data[i].GetStartTime().GetTime());
                binaryWriter.Write(data[i].GetEndTime().GetTime());
            }
            binaryWriter.Close();
        }

        public static Call[] ReadFile(string path)
        {
            List<Call> calls = new List<Call>();
            BinaryReader binaryReader = new BinaryReader(File.Open(path, FileMode.Open));
            while(binaryReader.PeekChar() > -1)
            {
                string phoneNumber = binaryReader.ReadString();
                Time startTime = new Time(binaryReader.ReadString());
                Time endTime = new Time(binaryReader.ReadString());

                Call call = new Call(phoneNumber, startTime, endTime);
                calls.Add(call);
            }
            binaryReader.Close();

            return calls.ToArray();
        }

        public static void DisplayFile(string path, string caption = "File: ")
        {
            BinaryReader binaryReader = new BinaryReader(File.Open(path, FileMode.Open));

            Console.WriteLine(caption);

            while (binaryReader.PeekChar() > -1)
            {
                string phoneNumber = binaryReader.ReadString();
                Time startTime = new Time(binaryReader.ReadString());
                Time endTime = new Time(binaryReader.ReadString());

                Console.WriteLine(phoneNumber + " " + startTime.GetTime() + " " + endTime.GetTime());
            }
            binaryReader.Close();
        }
    }
}
